src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2269,10 +2822,34 @@ private void onApplicationBegin(
                         AmqpReceiverSettleMode amqpReceiverSettleMode =
                             amqpReceiverSettleMode(amqpBeginEx.receiverSettleMode().get());
                         final String address = amqpBeginEx.address().asString();
+                        String otherAddress = null;
+                        switch (amqpRole)
+                        {
+                        case SENDER:
+                            otherAddress = addressTo;
+                            break;
+                        case RECEIVER:
+                            otherAddress = addressFrom;
+                            break;
+                        }
 
                         deliveryCount = initialDeliveryCount;
                         doEncodeAttach(traceId, authorization, name, channelId, handle, amqpRole, amqpSenderSettleMode,
-                            amqpReceiverSettleMode, address);
+                            amqpReceiverSettleMode, address, otherAddress);
```

Comment:
What if `amqpBeginEx` is `null` ?

Reply:
Would it be `amqp:decode-error` ?

Reply:
i think it just means we go with defaults as required by the protocol, since there was no issue with the decode that produced the initial amqp begin.